### PR TITLE
Change self.seed to self.random_state to pass check_constructor.

### PR DESCRIPTION
**Pull request by angus924**
#### Reference Issues/PRs

Fix estimators (#257)

#### What does this implement/fix? Explain your changes.

Change self.seed to self.random_state to pass check_constructor and test_estimator.

### CODE REVIEW

**Comment by mloning** [comment]
Hi @angus924, thanks for the fix! Could you please remove your estimator from [this list](https://github.com/alan-turing-institute/sktime/blob/20f524db48897e4ef0c02a9b2db08f42fa86a3f3/sktime/tests/config.py#L47)?

**Comment by angus924** [comment]
Done, sorry, oversight on my part.

File: sktime/transformers/series_as_features/rocket.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ class Rocket(BaseSeriesAsFeaturesTransformer):
     def __init__(self, num_kernels=10_000, normalise=True, random_state=None):
         self.num_kernels = num_kernels
         self.normalise = normalise
-        self.seed = random_state if isinstance(random_state, int) else None
+        self.random_state = \
```
Comment (mloning) [inline sktime/transformers/series_as_features/rocket.py:40]: Could we simply assign the random state here, whatever it is, and then you [`check_random_state`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html) from scikit-learn in the `_generate_kernels_function`?

Comment (mloning) [reply sktime/transformers/series_as_features/rocket.py:40]: Everything else looks good! 🙂

Comment (mloning) [reply sktime/transformers/series_as_features/rocket.py:40]: Please open a separate PR for the random state stuff. 

The `check_random_state` returns a random state object, so you can use that directly to use Numpy's random functionality, e.g.

rng = check_random_state(self.random_state)
subsample = rng.randint(self.n_instances, size=subsample_size)

Comment (angus924) [reply sktime/transformers/series_as_features/rocket.py:40]: So the reason I'm not doing this at the moment is that you can't pass a numpy random state object into numba compiled code.  I don't think there's a way around it at the moment.  It's possible to pass in an integer to seed numba's own random state (what I'm doing here), but that's it.

Comment (mloning) [reply sktime/transformers/series_as_features/rocket.py:40]: Okay, I remember talking about this before, let's leave it as it is :)
